### ISSUE TITLE
[Request] Libretro Core

### ISSUE THREAD

**Issue by SpaceAgeHero** [stenzek/duckstation#7]
I've noticed your frequent activity and the emulator seems to be shaping up quite well.
Would you also consider making a Libretro core too?
Through Retroarch I think it could be made accessible to many users on various platforms.

**Comment by stenzek** [comment]
No immediate plans but the emulator core and frontends are well seperated (there's currently an android, SDL and qt front-end), so it could be added without any core modifications.

**Comment by stenzek** [comment]
Preliminary libretro core is added as of https://github.com/stenzek/duckstation/commit/d7f083559eb4fb6d4cdd2c66daebd4d81ff60c53. Still needs some work, TODO from the source file:
  - Fix up D3D11
  - Save states
  - Expose the rest of the options
  - Memory card and controller settings
  - Better paths for memory cards/BIOS
  - Fix up aspect ratio

But, the basic stuff is there. Enable BUILD_LIBRETRO_CORE in cmake, or windows builds should happen automatically (not tested). When it's more feature complete I'll update the main README, but if anyone wants to play around with what's there, you can :)

**Comment by stenzek** [comment]
Closing now as it is now mostly feature complete. Only the Vulkan renderer and lightgun support remain.

Performance is comparable to the main DuckStation frontends (very slightly slower), but still significantly faster than Beetle HW (with its recompiler in max performance) and PCSX-Rearmed (also recompiler enabled). But I only tested a couple of games. FF7 for example runs at ~135fps in Beetle, ~460fps in pcsxr, and ~630fps in DuckStation.
